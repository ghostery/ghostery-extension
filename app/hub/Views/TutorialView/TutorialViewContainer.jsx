/**
 * Tutorial View Container
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2018 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

import React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import TutorialView from './TutorialView';
import * as actions from './TutorialViewActions';

// Component Views
import TutorialVideoView from '../TutorialViews/TutorialVideoView';
import TutorialTrackerListView from '../TutorialViews/TutorialTrackerListView';
import TutorialSimpleDetailedView from '../TutorialViews/TutorialSimpleDetailedView';
import TutorialBlockingView from '../TutorialViews/TutorialBlockingView';
import TutorialTrustRestrictView from '../TutorialViews/TutorialTrustRestrictView';
import TutorialAntiSuiteView from '../TutorialViews/TutorialAntiSuiteView';

/**
 * @class Implement the Tutorial View Container for the Ghostery Hub
 * @extends Component
 * @memberof HubContainers
 */
class TutorialViewContainer extends React.Component {
	/**
	 * Lifecycle Event
	 */
	componentWillMount() {
		const title = '';
		window.document.title = title;
	}

	/**
	 * React's required render function. Returns JSX
	 * @return {JSX} JSX for rendering the Tutorial View of the Hub app
	 */
	render() {
		const activeIndex = +this.props.location.pathname.split('/').pop();
		const steps = [
			{
				index: 1,
				path: '/tutorial/1',
				bodyComponent: TutorialVideoView,
			},
			{
				index: 2,
				path: '/tutorial/2',
				bodyComponent: TutorialTrackerListView,
			},
			{
				index: 3,
				path: '/tutorial/3',
				bodyComponent: TutorialSimpleDetailedView,
			},
			{
				index: 4,
				path: '/tutorial/4',
				bodyComponent: TutorialBlockingView,
			},
			{
				index: 5,
				path: '/tutorial/5',
				bodyComponent: TutorialTrustRestrictView,
			},
			{
				index: 6,
				path: '/tutorial/6',
				bodyComponent: TutorialAntiSuiteView,
			},
		];

		return <TutorialView activeIndex={activeIndex} steps={steps} />;
	}
}

// Default props used throughout the Tutorial flow
TutorialViewContainer.defaultProps = {};

/**
 * Map redux store state properties to the component's own properties.
 * @param  {Object} state    entire Redux store's state
 * @return {function}        this function returns a plain object, which will be merged into the component's props
 * @memberof HubContainers
 */
const mapStateToProps = () => Object.assign({});

/**
 * Bind the component's action creators using Redux's bindActionCreators.
 * @param  {function} dispatch redux store method which dispatches actions
 * @return {function}          to be used as an argument in redux connect call
 * @memberof SetupContainers
 */
const mapDispatchToProps = dispatch => ({
	actions: bindActionCreators(Object.assign(actions), dispatch),
});

export default connect(mapStateToProps, mapDispatchToProps)(TutorialViewContainer);
