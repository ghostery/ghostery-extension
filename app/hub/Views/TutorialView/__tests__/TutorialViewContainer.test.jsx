/**
 * Tutorial View Container Test Component
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2018 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router';

// Mock Necessary Imports
jest.mock('../../TutorialViews/TutorialVideoView', () => props => <div>Mock Tutorial Video View</div>);
jest.mock('../../TutorialViews/TutorialTrackerListView', () => props => <div>Mock Tutorial Tracker List View</div>);
jest.mock('../../TutorialViews/TutorialBlockingView', () => props => <div>Mock Tutorial Blocking View</div>);
jest.mock('../../TutorialViews/TutorialLayoutView', () => props => <div>Mock Tutorial Layout View</div>);
jest.mock('../../TutorialViews/TutorialTrustView', () => props => <div>Mock Tutorial Trust View</div>);
jest.mock('../../TutorialViews/TutorialAntiSuiteView', () => props => <div>Mock Tutorial Anti Suite View</div>);
jest.mock('../../TutorialViews/TutorialNavigation', () => props => <div>Mock Tutorial Navigation View</div>);

// Import Components
import TutorialViewContainer from '../TutorialViewContainer';

// Fake Actions
const actions = {
	initTutorialProps: () => {
		return new Promise((resolve) => {
			resolve();
		});
	},
	setTutorialNavigation: () => {},
};

describe('app/hub/Views/TutorialView container', () => {
	describe('Snapshot tests with react-test-renderer', () => {
		test('tutorial view container is rendered correctly on the tutorial video step', () => {
			const paths = ['/tutorial/1', '/tutorial/2', '/tutorial/3', '/tutorial/4', '/tutorial/5', '/tutorial/6'];
			const component = renderer.create(
				<MemoryRouter initialEntries={paths} initialIndex={0} >
					<TutorialViewContainer actions={actions} preventRedirect={true} />
				</MemoryRouter>
			).toJSON();
			expect(component).toMatchSnapshot();
		});

		test('tutorial view container is rendered correctly on the tutorial tracker list step', () => {
			const paths = ['/tutorial/1', '/tutorial/2', '/tutorial/3', '/tutorial/4', '/tutorial/5', '/tutorial/6'];
			const component = renderer.create(
				<MemoryRouter initialEntries={paths} initialIndex={1} >
					<TutorialViewContainer actions={actions} preventRedirect={true} />
				</MemoryRouter>
			).toJSON();
			expect(component).toMatchSnapshot();
		});

		test('tutorial view container is rendered correctly on the tutorial blocking step', () => {
			const paths = ['/tutorial/1', '/tutorial/2', '/tutorial/3', '/tutorial/4', '/tutorial/5', '/tutorial/6'];
			const component = renderer.create(
				<MemoryRouter initialEntries={paths} initialIndex={2} >
					<TutorialViewContainer actions={actions} preventRedirect={true} />
				</MemoryRouter>
			).toJSON();
			expect(component).toMatchSnapshot();
		});

		test('tutorial view container is rendered correctly on the tutorial layout step', () => {
			const paths = ['/tutorial/1', '/tutorial/2', '/tutorial/3', '/tutorial/4', '/tutorial/5', '/tutorial/6'];
			const component = renderer.create(
				<MemoryRouter initialEntries={paths} initialIndex={3} >
					<TutorialViewContainer actions={actions} preventRedirect={true} />
				</MemoryRouter>
			).toJSON();
			expect(component).toMatchSnapshot();
		});

		test('tutorial view container is rendered correctly on the tutorial trust and restrict step', () => {
			const paths = ['/tutorial/1', '/tutorial/2', '/tutorial/3', '/tutorial/4', '/tutorial/5', '/tutorial/6'];
			const component = renderer.create(
				<MemoryRouter initialEntries={paths} initialIndex={4} >
					<TutorialViewContainer actions={actions} preventRedirect={true} />
				</MemoryRouter>
			).toJSON();
			expect(component).toMatchSnapshot();
		});

		test('tutorial view container is rendered correctly on the tutorial anti suite step', () => {
			const paths = ['/tutorial/1', '/tutorial/2', '/tutorial/3', '/tutorial/4', '/tutorial/5', '/tutorial/6'];
			const component = renderer.create(
				<MemoryRouter initialEntries={paths} initialIndex={5} >
					<TutorialViewContainer actions={actions} preventRedirect={true} />
				</MemoryRouter>
			).toJSON();
			expect(component).toMatchSnapshot();
		});
	});
});
