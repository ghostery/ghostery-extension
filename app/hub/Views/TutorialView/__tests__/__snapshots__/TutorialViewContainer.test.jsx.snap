// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app/hub/Views/TutorialView container Snapshot tests with react-test-renderer tutorial view container is rendered correctly on the tutorial anti suite step 1`] = `
<div
  className="full-height flex-container flex-dir-column"
>
  <div
    className="TutorialView--paddingTopSmall flex-child-grow flex-container align-middle"
  >
    <div>
      Mock Tutorial Anti Suite View
    </div>
  </div>
  <div>
    Mock Tutorial Navigation View
  </div>
</div>
`;

exports[`app/hub/Views/TutorialView container Snapshot tests with react-test-renderer tutorial view container is rendered correctly on the tutorial blocking step 1`] = `
<div
  className="full-height flex-container flex-dir-column"
>
  <div
    className="TutorialView--paddingTopSmall flex-child-grow flex-container align-middle"
  >
    <div>
      Mock Tutorial Blocking View
    </div>
  </div>
  <div>
    Mock Tutorial Navigation View
  </div>
</div>
`;

exports[`app/hub/Views/TutorialView container Snapshot tests with react-test-renderer tutorial view container is rendered correctly on the tutorial layout step 1`] = `
<div
  className="full-height flex-container flex-dir-column"
>
  <div
    className="TutorialView--paddingTopSmall flex-child-grow flex-container align-middle"
  >
    <div>
      Mock Tutorial Layout View
    </div>
  </div>
  <div>
    Mock Tutorial Navigation View
  </div>
</div>
`;

exports[`app/hub/Views/TutorialView container Snapshot tests with react-test-renderer tutorial view container is rendered correctly on the tutorial tracker list step 1`] = `
<div
  className="full-height flex-container flex-dir-column"
>
  <div
    className="TutorialView--paddingTopSmall flex-child-grow flex-container align-middle"
  >
    <div>
      Mock Tutorial Tracker List View
    </div>
  </div>
  <div>
    Mock Tutorial Navigation View
  </div>
</div>
`;

exports[`app/hub/Views/TutorialView container Snapshot tests with react-test-renderer tutorial view container is rendered correctly on the tutorial trust and restrict step 1`] = `
<div
  className="full-height flex-container flex-dir-column"
>
  <div
    className="TutorialView--paddingTopSmall flex-child-grow flex-container align-middle"
  >
    <div>
      Mock Tutorial Trust View
    </div>
  </div>
  <div>
    Mock Tutorial Navigation View
  </div>
</div>
`;

exports[`app/hub/Views/TutorialView container Snapshot tests with react-test-renderer tutorial view container is rendered correctly on the tutorial video step 1`] = `
<div
  className="full-height flex-container flex-dir-column"
>
  <div
    className="TutorialView--paddingTopSmall flex-child-grow flex-container align-middle"
  >
    <div>
      Mock Tutorial Video View
    </div>
  </div>
  <div>
    Mock Tutorial Navigation View
  </div>
</div>
`;
